core/domain/opportunity_services.py

```diff
@@ -850,7 +850,8 @@ def _get_skill_opportunities_with_updated_question_counts(skill_ids, delta):
         if skill_opportunity_model is not None:
             skill_opportunity = get_skill_opportunity_from_model(
                 skill_opportunity_model)
-            skill_opportunity.question_count += delta
+            skill_opportunity.question_count = max(
+                skill_opportunity.question_count + delta, 0)
```

Comment:
What does this do?

Reply:
Here I'm defaulting to 0 if some operation tries to reduce question count to a negative value. Added a comment in the code.